app/routers/sessions.py

```diff
@@ -274,10 +330,17 @@ async def update_session(session_id: str, session_updates: UpdateSession):
 
     # update the document in the sessions collection
     if new_event == EventType.end_quiz:
+        session_metrics = reformat_metrics(jsonable_encoder(session_updates)["metrics"])
```

Comment:
one thing I didn't understand is why are we doing this reformatting / extra processing. Isn't everything already calculated on the frontend and should be just sent to the DB through the backend?
Also worried about these extra processing because if its unnecessary computation then it will put load on the system when its under scale. Either if its lambda or ec2 server.

Reply:
removed reformatting. shifted to frontend